sssd_test_framework/utils/sssd.py

```diff
@@ -360,21 +359,27 @@ def set_service_user(self, user: str) -> None:
 
         service_file = "/usr/lib/systemd/system/sssd.service"
         rw = self.host.conn.run(f"test -w {service_file}", raise_on_error=False)
+        cmd = ""
         if rw.rc != 0:
-            # Can't change service user on read only filesystem
-            pytest.skip(f"Cannot write {service_file}")
-
-        self.fs.backup(service_file)
-        cmd = f'sed -i "s/^User=.*/User={user}/g" {service_file}\n'
-        cmd += f'sed -i "s/^Group=.*/Group={user}/g" {service_file}\n'
-        if user == "root":
-            cmd += f'sed -i "s/^#SupplementaryGroups=sssd$/SupplementaryGroups=sssd/g" {service_file}\n'
-            cmd += f'sed -i "s/sssd:sssd/root:root/g" {service_file}\n'
-        elif user == "sssd":
-            cmd += f'sed -i "s/^SupplementaryGroups=sssd$/#SupplementaryGroups=sssd/g" {service_file}\n'
-            cmd += f'sed -i "s/root:root/sssd:sssd/g" {service_file}\n'
+            self.logger.info(f"Using service overlay on read only filesystem on {self.host.hostname}")
+            self.fs.mkdir("/etc/systemd/system/sssd.service.d")
+            supp = "SupplementaryGroups=sssd\n" if user == "root" else ""
+            contents = f"[Service]\nUser={user}\nGroup={user}\n{supp}"
+            self.fs.write(
+                "/etc/systemd/system/sssd.service.d/user.conf", contents, mode="0600", user="root", group="root"
+            )
         else:
-            raise ValueError("Unexpected value of 'user'")
+            self.fs.backup(service_file)
+            cmd += f'sed -i "s/^User=.*/User={user}/g" {service_file}\n'
+            cmd += f'sed -i "s/^Group=.*/Group={user}/g" {service_file}\n'
+            if user == "root":
+                cmd += f'sed -i "s/^#SupplementaryGroups=sssd$/SupplementaryGroups=sssd/g" {service_file}\n'
+                cmd += f'sed -i "s/sssd:sssd/root:root/g" {service_file}\n'
```

Comment:
This edits args of 'chown' in main file. This is missing in override.

Reply:
Added extra code for this and trying again.

Reply:
I don't need.
This needs to be tested.
Could you please capture output of `systemctl cat sssd` both for 'sssd -> root' and 'root -> sssd' transition?

Reply:
We need to see resulting service file to make sure the order is right (i.e. override's 'ExecStartPre' go *after* original content):
```
Multiple command lines are allowed, regardless of the service type (i.e. Type=),
and the commands are executed one after the other, serially.
```

Reply:
The override was shown after the original unit wnen using systecmtcl cat